site/pages/retirement/[beneficiary_address]/[retirement_index].tsx

```diff
@@ -0,0 +1,64 @@
+import { GetStaticProps } from "next";
+
+import { queryKlimaRetireByIndex } from "@klimadao/lib/utils";
+import { SingleRetirementPage } from "components/pages/Retirement/SingleRetirement";
+import { loadTranslation } from "lib/i18n";
+import { IS_PRODUCTION } from "lib/constants";
+
+export const getStaticProps: GetStaticProps = async (ctx) => {
+  try {
+    if (IS_PRODUCTION) {
+      throw new Error("Not on Staging");
+    }
+
+    const { params, locale } = ctx;
+
+    if (
+      !params ||
+      (!params?.beneficiary_address && !params?.retirement_index) ||
+      (!!params.retirement_index && Number(params.retirement_index) === NaN)
+    ) {
+      throw new Error("No params found");
+    }
+
+    const retirementIndex = Number(params.retirement_index) - 1; // totals does not include index 0
+
+    const retirement = await queryKlimaRetireByIndex(
+      params?.beneficiary_address as string,
+      retirementIndex
+    );
```

Comment:
have you done some test retires to see how quickly this subgraph updates? As we said on the call this could have a delay, which was why I assumed we'd have to rely on the contracts...

Reply:
Just tested it again:

- doing an offset here: https://app.klimadao.finance/#/offset
- loading the page on this PR => immediately generated